tests/subsys/secure_storage/psa/its/src/main.c

```diff
@@ -4,7 +4,49 @@
 #include <zephyr/ztest.h>
 #include <psa/internal_trusted_storage.h>
 
+#include <zephyr/types.h>
+#include <zephyr/storage/flash_map.h>
+#include <zephyr/drivers/flash.h>
+
 /* The flash must be erased after this test suite is run for the write-once entry test to pass. */
+
+#if !defined(CONFIG_BUILD_WITH_TFM) && defined(CONFIG_FLASH_HAS_EXPLICIT_ERASE)
+
+#define MAX_NUM_PAGES 2
+static const struct device *const fdev = DEVICE_DT_GET(DT_CHOSEN(zephyr_flash_controller));
+
+#define FLASH_ERASE_END_ADDR                                                                       \
+	(FIXED_PARTITION_OFFSET(storage_partition) + FIXED_PARTITION_SIZE(storage_partition))
+
+static void erase_flash(void)
+{
+	int rc;
+	off_t address = FIXED_PARTITION_OFFSET(storage_partition);
+	struct flash_pages_info page_info;
+
+#if defined(CONFIG_FLASH_HAS_NO_EXPLICIT_ERASE)
+	const struct flash_parameters *fparam = flash_get_parameters(fdev);
+
+	if (!(flash_params_get_erase_cap(fparam) & FLASH_ERASE_C_EXPLICIT)) {
```

Comment:
Maybe in case we have no erase operation (e.g. nRF54 RRAM and MRAM) we should fully overwrite the region with zeroes or ones? That would also erase the data to make sure we start from a known-empty state.

native_sim can also simulate such NVM model IIRC, so it should be testable even without such HW.